### PR TITLE
Compile `tocvdebug.d` for all platforms

### PR DESCRIPTION
**Pull request by thewilsonator**
Make DMD's anemic cross-compilation, slightly less anemic.

rebase of https://github.com/dlang/dmd/pull/12693

### CODE REVIEW

File: src/dmd/tocvdebug.d [hunk 1]
```diff
@@ -309,6 +304,9 @@ void cv_udt(const char* id, uint typidx)
 
 void toDebug(EnumDeclaration ed)
 {
+    if (target.os != Target.OS.Windows)
+             return;
```
Comment (dkorpel) [inline src/dmd/tocvdebug.d:308]: The indentation of these `return` statements is off

File: src/dmd/tocvdebug.d [hunk 2]
```diff
@@ -54,8 +51,6 @@ import dmd.backend.oper;
 import dmd.backend.ty;
 import dmd.backend.type;
 
-extern (C++):
```
Comment (dkorpel) [inline src/dmd/tocvdebug.d:57]: Is this related?

Comment (ibuclaw) [reply src/dmd/tocvdebug.d:57]: This is a dmd-only backend module, I can't see any reason why these must be exported to C++.

Comment (dkorpel) [reply src/dmd/tocvdebug.d:57]: Okay, I just wondered if removing it was needed for cross compilation

File: src/dmd/tocvdebug.d (hunk 1 above)
Comment (thewilsonator) [reply src/dmd/tocvdebug.d]: Fixed.

**Review by dkorpel — APPROVED** [review]

File: src/dmd/tocvdebug.d (hunk 2 above)
Comment (thewilsonator) [reply src/dmd/tocvdebug.d:57]: No I was just cleaning out old cruft.
